src/averageAges.js

```diff
@@ -37,7 +45,13 @@ function calculateMenAverageAge(people, century) {
  * @return {number}
  */
 function calculateWomenAverageAge(people, withChildren) {
-  // write code here
+  const women = people.filter(person =>
+    withChildren
+      ? person.sex === 'f' && people.some(child => child.mother === person.name)
```

Comment:
```suggestion
      ? people.some(child => child.mother === person.name)
```
actually, if person is somebody's mom, no need to check her sex